evm/src/cpu/decode.rs

```diff
@@ -184,15 +157,19 @@ pub fn generate<F: RichField>(lv: &mut CpuColumnsView<F>) {
     assert!(kernel <= 1);
     let kernel = kernel != 0;
 
-    for (oc, block_length, availability, col) in OPCODES {
-        let available = match availability {
-            All => true,
-            User => !kernel,
-            Kernel => kernel,
-        };
+    let mut any_flag_set = false;
+    for (oc, block_length, kernel_only, col) in OPCODES {
+        let available = !kernel_only || kernel;
         let opcode_match = top_bits[8 - block_length] == oc;
-        lv[col] = F::from_bool(available && opcode_match);
+        let flag = available && opcode_match;
+        lv[col] = F::from_bool(flag);
+        if flag && any_flag_set {
+            panic!("opcode matched multiple flags");
+        }
+        any_flag_set = any_flag_set || flag;
```

Comment:
nit: could use `||=`

Reply:
That was my original attempt, but it turns out that `||=` is not a Rust operator: https://doc.rust-lang.org/reference/expressions/operator-expr.html#compound-assignment-expressions